### PR TITLE
dns-cloudflare: Update docs and error messages to reflect new API permissions

### LINKED ISSUE #7893: dns-cloudflare requires too many permissions
The `dns-cloudflare` module should provide support for manually specifying the `zone_id`. This quote from the documentation is well known:

> However, due to some shortcomings in Cloudflare’s implementation of Tokens, Tokens created for Certbot currently require Zone:Zone:Read and Zone:DNS:Edit permissions for **all** zones in your account. While this is not ideal, your Token will still have fewer permission than the Global key, so it’s still worth doing. Hopefully Cloudflare will improve this in the future.

You do not really need to give the token rights for **ALL** zones. You can successfully generate certificates with cloudflare if you use `manual-auth-hook`:
```
    certbot certonly \
      --manual \
      --non-interactive \
      --preferred-challenges dns \
      --manual-auth-hook "{{ work_folder }}/letsencrypt_auth.sh" \
      --manual-cleanup-hook "{{ work_folder }}/letsencrypt_cleanup.sh" \
      --rsa-key-size 4096 \
      --agree-tos \
      --manual-public-ip-logging-ok \
      --no-eff-email \
      --email {{ letsencrypt_email }} \
      --domains "*.{{ fqdn }}"
```
And having the following in your `letsencrypt_auth.sh`:
```
#!/bin/bash
EMAIL="xxx"
ZONE_ID="xxx"
TYPE="TXT"
NAME="_acme-challenge.$( echo "$CERTBOT_DOMAIN" | cut -d"." -f1-2 )"
CONTENT="$CERTBOT_VALIDATION"
TTL="1"
curl -X POST "https://api.cloudflare.com/client/v4/zones/$ZONE_ID/dns_records/" \
    -H "Authorization: xxxxxxxx" \
    -H "Content-Type: application/json" \
    --data '{"type":"'"$TYPE"'","name":"'"$NAME"'","content":"'"$CONTENT"'","ttl":'"$TTL"'}' 
```

When running with `certbot certonly --dns-cloudflare`, you will get an error as follows:
> Unable to determine zone_id for xxx using zone names: ['xxx', 'xxx', 'xxx', 'xxx']. Please confirm that the domain name has been entered correctly and is already associated with the supplied Cloudflare account. The error from Cloudflare was: 0 Actor 'com.cloudflare.api.token.xxxx' requires permission 'com.cloudflare.api.account.zone.list' to list zones

This happens because certbot forces you to give it access to all the zones under the account, instead of allowing you to manually specify the zone id.

### PR DESCRIPTION
**Pull request by Tugzrida**
Cloudflare has now fixed a bug in their API which required the API token for Certbot to have quite wide-ranging permissions. This PR updates the docs to reflect the fix, updates an error code and also improves the link to the Cloudflare dashboard with deep linking.

By my testing, tokens now only need `Zone:DNS:Edit` for whichever zone a cert is required for.

Through every combination of incorrect credentials and permissions I could think of, there's only a small issue remaining with one of the error messages:

certbot.errors.PluginError: Unable to determine zone_id for example.com using zone names: ['example.com', 'com']. Please confirm that the domain name has been entered correctly and is already associated with the supplied Cloudflare account. The error from Cloudflare was: 0 Actor 'com.cloudflare.api.token.' requires permission 'com.cloudflare.api.account.zone.list' to list zones

If a Global Key is used with a non-existent domain, this error(without the "The error from Cloudflare was: ..." part) shows, which is fine.

If using a Token, the (full) error shows with both a non-existent domain _or a domain that the Token doesn't have access to_.

I'm not sure what the best way to tidy this up is, as it's already quite long. I was thinking of replacing the "The error from Cloudflare was: ..." part with something like "Please confirm that your Cloudflare Token has access to the domain.", however that is the only place that unrecognised errors from Cloudflare are printed, because CF is still returning an error (unhelpfully with the non-unique code of 0) when we test the incorrect domain guesses, and so errors are only printed when no matches are found from all guesses.

Closes #7893

cc @bmw @adferrand

### CODE REVIEW

**Comment by bmw** [comment]
Thanks again @Tugzrida.

I'll try to review this in the next couple weeks.

File: certbot-dns-cloudflare/certbot_dns_cloudflare/_internal/dns_cloudflare.py [hunk 1]
```diff
@@ -118,7 +118,7 @@ def add_txt_record(self, domain, record_name, record_content, record_ttl):
             code = int(e)
             hint = None
 
-            if code == 9109:
+            if code == 10000:
```
[the remark this answers is not in the document]
Comment (Tugzrida) [reply certbot-dns-cloudflare/certbot_dns_cloudflare/_internal/dns_cloudflare.py:121]: I can't get 10000 anymore either, only 1009. 10000 is not a documented error code for that endpoint so I'm guessing it was an error on Cloudflare's end.

**Comment by Tugzrida** [comment]
Filtering on the content of the error message was the only way I could see to get a neat result, but I was also hesitant to implement that. The way it is now, if that string isn't found, the error message will just be a bit less specific.

Please let me know if this is too messy, but the error message is now as follows:

If the token doesn't have access to the domain:
`certbot.errors.PluginError: Unable to determine zone_id for example.com using zone names: ['example.com', 'com']. Please confirm that the domain name has been entered correctly and your Cloudflare Token has access to the domain.`

If a global key can't access the domain:
`certbot.errors.PluginError: Unable to determine zone_id for example.com using zone names: ['example.com', 'com']. Please confirm that the domain name has been entered correctly and is already associated with the supplied Cloudflare account.`

Any other error occurred:
`certbot.errors.PluginError: Unable to determine zone_id for example.com using zone names: ['example.com', 'com']. The error from Cloudflare was: 0 connection failed.`

I added this third case instead of just logging the Cloudflare error, as users would need to see other errors such as a connection error, etc.

**Comment by bmw** [comment]
This LGTM. Thanks again for all your help here @Tugzrida.

@adferrand, if you still have the time and interest to give this a 2nd review when you get the chance, that'd be helpful.

**Pull request merged**